mathesar/views/api.py

```diff
@@ -331,19 +331,31 @@ def create(self, request, table_pk=None):
         table = get_table_or_404(table_pk)
         # We only support adding a single record through the API.
         assert isinstance((request.data), dict)
-        record = table.create_record_or_records(request.data)
-        serializer = RecordSerializer(record)
+        try:
+            record = table.create_record_or_records(request.data)
+            serializer = RecordSerializer(record)
+        except NotUniquePrimaryKey as e:
+            logger.error(e.message)
+            raise e
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
     def partial_update(self, request, pk=None, table_pk=None):
         table = get_table_or_404(table_pk)
-        record = table.update_record(pk, request.data)
-        serializer = RecordSerializer(record)
+        try:
+            record = table.update_record(pk, request.data)
+            serializer = RecordSerializer(record)
+        except NotUniquePrimaryKey as e:
```

Comment:
Please write additional tests for retrieve, deletion, and partial updates that ensure this error is raised (for both no primary keys and multiple primary keys)